lib/bolt/config.rb

```diff
@@ -224,14 +237,21 @@ def update_from_file(data)
       @trusted_external = if data.key?('trusted-external-command')
                             File.expand_path(data['trusted-external-command'], @boltdir.path)
                           end
+
+      if data.key?('apply_settings')
+        @apply_settings, extra_opts = data['apply_settings'].partition do |k, _|
+          APPLY_SETTINGS.keys.include?(k)
+        end.map(&:to_h)
+        unless extra_opts.empty?
+          @logger.warn("Apply settings #{extra_opts.keys.join(', ')} are not valid in Bolt")
```

Comment:
This warning will never get displayed since the logger isn't configured until after the config.